presto-main/src/test/java/com/facebook/presto/operator/scalar/TestZipTransformFunction.java

```diff
@@ -0,0 +1,100 @@
+
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.RowType;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.operator.scalar.ZipFunction.MAX_ARITY;
+import static com.facebook.presto.operator.scalar.ZipFunction.MIN_ARITY;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+
+public class TestZipTransformFunction
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testSameLength()
+            throws Exception
+    {
+        assertFunction("zip(ARRAY[1, 2], ARRAY['a', 'b'], (x, y) -> (y, x))",
+                zipReturnType(createVarcharType(1), INTEGER),
+                list(list("a", 1), list("b", 2)));
+
+        assertFunction("zip(ARRAY[1, 2], ARRAY[CAST('a' AS VARCHAR), CAST('b' AS VARCHAR)], (x, y) -> (y, x))",
+                zipReturnType(VARCHAR, INTEGER),
+                list(list("a", 1), list("b", 2)));
+
+        assertFunction("zip(ARRAY[], ARRAY[], (x, y) -> (y, x))",
+                zipReturnType(UNKNOWN, UNKNOWN),
+                list());
+
+        assertFunction("zip(ARRAY[NULL], ARRAY[NULL], (x, y) -> (y, x))",
+                zipReturnType(UNKNOWN, UNKNOWN),
+                list(list(null, null)));
+
+        assertFunction("zip(ARRAY[1, 1], ARRAY[1, 2], (x, y) -> x + y)",
+                new ArrayType(INTEGER),
+                list(2, 3));
+    }
+
+    @Test
+    public void testDifferentLength()
+            throws Exception
+    {
+        assertFunction("zip(ARRAY[1], ARRAY['a', 'b'], (x, y) -> (y, x))",
+                zipReturnType(createVarcharType(1), INTEGER),
+                list(list("a", 1), list("b", null)));
+
+        assertFunction("zip(ARRAY[NULL, 2], ARRAY['a'], (x, y) -> (y, x))",
+                zipReturnType(createVarcharType(1), INTEGER),
+                list(list("a", null), list(null, 2)));
+
+        assertFunction("zip(ARRAY[1, NULL], ARRAY[3, 2, 1], (x, y) -> x + y)",
+                new ArrayType(INTEGER),
+                list(4, null, null));
+    }
+
+    @Test
+    public void testWithNull()
+            throws Exception
+    {
+        assertFunction("zip(CAST(NULL AS ARRAY(UNKNOWN)), ARRAY[], (x, y) -> (y, x))",
+                zipReturnType(UNKNOWN, UNKNOWN),
+                null);
+    }
+
+    private static Type zipReturnType(Type... types)
+    {
+        return new ArrayType(new RowType(list(types), Optional.empty()));
+    }
+
+    @SafeVarargs
+    private static <T> List<T> list(T... a)
```

Comment:
Use `ImmutableList.of` whenever possible. When `null` element is present, use `Arrays.asList`. It's not necessary to introduce this `list` function.